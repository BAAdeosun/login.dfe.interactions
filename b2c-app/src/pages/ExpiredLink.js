import React from 'react';
import components from '../components';
import { ACTIONS } from '../constants/actions';
import { LINK_TYPES } from '../constants/linkTypes';

class ExpiredLink extends React.Component {

    render() {

        let linkExpiredParagraph;
        let requestAgainLink;

        //we keep the logic to check different actions passed in, in case we have to reuse this page for different flows again
        if (this.props.action === ACTIONS.RESET_PASSWORD) {
            linkExpiredParagraph =
                <components.Paragraph>
                    The link in your password reset email has expired.
                </components.Paragraph>

            requestAgainLink =
                <components.Link type={LINK_TYPES.BUTTON} action={ACTIONS.RESET_PASSWORD}>Resend password reset email</components.Link>
        }

        const content =
            <div>
                {linkExpiredParagraph}
                <components.Paragraph>
                    Request another email.
                </components.Paragraph>
                {requestAgainLink}
            </div>

        const title = 'Expired link';

        const pageConfig = {
            title: title,
            header: title,
            aboveFormContent: content
        };


        return (
            <div id="expiredLink">
                <components.PageContainer pageConfig={pageConfig} />
            </div>
        )
    }
}

export default ExpiredLink;
