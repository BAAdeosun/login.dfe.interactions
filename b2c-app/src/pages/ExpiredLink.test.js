import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ExpiredLink from './ExpiredLink';

it('renders without crashing', () => {
    shallow(<ExpiredLink />);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<ExpiredLink />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});