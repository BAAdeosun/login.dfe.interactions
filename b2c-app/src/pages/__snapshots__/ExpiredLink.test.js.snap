// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with policy POLICIES.RESET_PASSWORD passed in 1`] = `
<div
  id="expiredLink"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              Expired link
            </h1>
            <div>
              <p
                className="govuk-body"
              >
                The link in your password reset email has expired.
              </p>
              <a
                className="govuk-button govuk-button--start"
                data-module="govuk-button"
                draggable="false"
                href={null}
                onClick={null}
                role="button"
              >
                Resend password reset email
              </a>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;

exports[`renders correctly without policy passed in 1`] = `
<div
  id="expiredLink"
>
  <div>
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-breadcrumbs"
      >
        <ol
          className="govuk-breadcrumbs__list"
        >
          <li
            className="govuk-breadcrumbs__list-item"
          >
            <a
              className="govuk-breadcrumbs__link"
              href="https://nationalcareers.service.gov.uk/"
            >
              Home
            </a>
          </li>
        </ol>
      </div>
      <main
        className="govuk-main-wrapper"
      >
        <div
          className="govuk-grid-row"
        >
          <div
            className="govuk-grid-column-two-thirds"
          >
            <h1
              className="govuk-heading-xl govuk-!-margin-top-6 govuk-!-margin-bottom-9"
            >
              Expired link
            </h1>
            <div>
              <p
                className="govuk-body"
              />
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
